client/auth.go

```diff
@@ -0,0 +1,386 @@
+package client
+
+import (
+	"crypto/rand"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+	xlineapi "github.com/xline-kv/go-xline/api/xline"
+	"golang.org/x/crypto/pbkdf2"
+)
+
+// Client for Auth operations.
+type authClient struct {
+	// Name of the AuthClient, which will be used in CURP propose id generation
+	name string
+	// The client running the CURP protocol, communicate with all servers.
+	curpClient curpClient
+	// The auth token
+	token string
+}
+
+// Creates a new `AuthClient`
+func newAuthClient(name string, curpClient curpClient, token string) authClient {
+	return authClient{
+		name:       name,
+		curpClient: curpClient,
+		token:      token,
+	}
+}
+
+// Enables authentication.
+func (c *authClient) AuthEnable() (*xlineapi.AuthEnableResponse, error) {
+	requestWithToken := xlineapi.RequestWithToken{
+		Token: &c.token,
+		RequestWrapper: &xlineapi.RequestWithToken_AuthEnableRequest{
+			AuthEnableRequest: &xlineapi.AuthEnableRequest{},
+		},
+	}
+	resp, err := c.handleReq(&requestWithToken, false)
+	if err != nil {
+		return nil, err
+	}
+	res := resp.CommandResp.GetAuthEnableResponse()
+	res.Header.Revision = resp.SyncResp.Revision
```

Comment:
After https://github.com/xline-kv/Xline/pull/391, the revision is no longer needs to be acquired from sync response.
I think we could use fast path here and the methods below. 

@Phoenix500526 @themanforfree  @iGxnon Please check if we could made this change.

Reply:
It seems like we've marked everything related to writing as a slow_path in the auth server. If this revision can be generated in advance, can they be changed to a fast_path?

Reply:
A deeper-level question: How to determine that a command must go through the slow_path?

Reply:
> A deeper-level question: How to determine that a command must go through the slow_path?

The key point of this question is whether the order of different commands matters or not. I'm afraid that we cannot put the auth operation in the fast-path phase since the order does matter in this scenario.  What do you say? @themanforfree @rogercloud

Reply:
What's the meaning of "putting auth in the fast-path phase"? Is it just put the auth command into speculative pool? I think auth operations conflict with all others. It prevents all the later operations from completing the fast-path.

Reply:
Makes sense. We only need to remove the revision assignment here.

Reply:
It looks like we're still going to use slow path, because auth and the other commands have to be in a strictly guaranteed order with each other.